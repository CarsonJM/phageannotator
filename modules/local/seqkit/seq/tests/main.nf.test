nextflow_process {

    name "Test Process: SEQKIT_SEQ"
    script "../main.nf"
    process "SEQKIT_SEQ"

    test("['modules_local']['assembly_fasta_gz']") {

        when {
            params {
                outdir              = "$outputDir"
                assembly_min_length = 5
            }
            process {
                """
                input[0] = [ [id: 'test'], [file(params.test_data['modules_local']['assembly_fasta_gz'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("['modules_local']['assembly_fasta_gz'] - stub") {

        options "-stub"

        when {
            params {
                outdir              = "$outputDir"
                assembly_min_length = 5
            }
            process {
                """
                input[0] = [ [id: 'test'], [file(params.test_data['modules_local']['assembly_fasta_gz'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
