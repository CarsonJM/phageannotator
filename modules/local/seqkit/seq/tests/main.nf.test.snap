{
    "['modules_local']['assembly_fasta_gz'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:52:51.168265662"
    },
    "['modules_local']['assembly_fasta_gz']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,5c3970a6403ad9383b04e0e71ed69ac7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ],
                "fastx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta.gz:md5,5c3970a6403ad9383b04e0e71ed69ac7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de038a980ca1e484c0e34f4af3f05c2a"
                ]
            }
        ],
        "timestamp": "2023-12-06T15:25:04.913990939"
    }
}